### ISSUE TITLE
Hi! I tested the ctag-juce-sampler!

### ISSUE THREAD

**Issue by ANGR77** [NiklasWan/CTAG-JUCE-Sampler#1]
And with the latest version of the Juce framework and compiled it using Visual Studio 2019. 

But when compiling I get a couple of errors. 6-7 errors...indicating on data types for some variables (like sound) in the Juce framework have changed from Auto to something else,,,

I am not too skilled on the Juce framework so I have not been digging it down yet to understand...but I felt it could be good to know! 

Best regards

Anders 

The errors I get when compiling: 
Severity	Code	Description	Project	File	Line	Suppression State	Suppression State

- Error (active)	E1587	cannot deduce 'auto' type	JUCESampler_SharedCode	C:\Users\ANGRO\Documents\CTAG-JUCE-Sampler-master\Source\CTAGSampler.cpp	64		

- Error	C3535	cannot deduce type for 'auto *' from 'juce::SynthesiserSound::Ptr'	JUCESampler_SharedCode	C:\Users\ANGRO\Documents\CTAG-JUCE-Sampler-master\Source\CTAGSampler.cpp	64		

- Error	C2440	'initializing': cannot convert from 'juce::SynthesiserSound::Ptr' to 'int *'	JUCESampler_SharedCode	C:\Users\ANGRO\Documents\CTAG-JUCE-Sampler-master\Source\CTAGSampler.cpp	64		

- Error	C2664	'void juce::Synthesiser::startVoice(juce::SynthesiserVoice *,juce::SynthesiserSound *,int,int,float)': cannot convert argument 2 from 'int *' to 'juce::SynthesiserSound *'	JUCESampler_SharedCode	C:\Users\ANGRO\Documents\CTAG-JUCE-Sampler-master\Source\CTAGSampler.cpp	119		

- Error	LNK1104	cannot open file 'JUCESampler.lib'	JUCESampler_StandalonePlugin	C:\Users\ANGRO\Documents\CTAG-JUCE-Sampler-master\Builds\VisualStudio2019\LINK	1

**Comment by NiklasWan** [comment]
Hey Anders,

thank you very much for reaching out. I'm sorry that it took so long but the issue is now resolved.
It seems like they changed the Implementation of Synthesiser::getSound() to return a SynthesiserSound::Ptr object instead of a raw pointer. Calling the SynthesiserSound::Ptr::get() method provides the raw pointer.

Cheers,
Niklas

**Comment by ANGR77** [comment]
Big Big Thanks for the fix, Niklas! (It is extremely hard to figure out things like this - if you are new to Juce as me...) :-)

Now, I got it running using VS2019 + Juce + Producer...a good example how to do it...Thanks!!!

FYI:
I was tempted to create a similar VST player but instead of playing 1 drum sound like you do for a snare...I have sound files with 4 dimensions (Mic positions) of the same snare sound. My idea was to create a mixer with a all 4 mic positions...let the app play the 4 mic positions at the same time. 
(I have created some professional sampled drum sounds for 2box this way...but was tempted to release them as a VST/AU as well... :-) 
:-) We will see if I succeed...but now I have a working demo of your nice code! Thanks again! 

Best Regards

Anders Gronlund
